### ISSUE TITLE
TLS configurations with self-hosted Cloud Pricing API

### ISSUE THREAD

**Issue by alikhajeh1** [infracost/infracost#941]
Self-hosted Cloud Pricing API users of GitHub Actions (or other CI/CD integrations), need a way to:
1. Use something like `INFRACOST_TLS_VERIFY=false` (similar to what `docker` supports) to disable TLS verification for testing. Given users went through the trouble of generating a cert, this option only seems useful for testing.
2. Have env variables or flags or `configure` keys for the following so users could provide the cert files (again, similar to what docker supports). On CI/CD, users would have to have a step in the GH Action that writes the cert files before Infracost runs.
```
      --tls-ca-cert string   Trust certs signed only by this CA (default "/Users/ali/.docker/ca.pem")
      --tls-cert string     Path to TLS certificate file (default "/Users/ali/.docker/cert.pem")
      --tls-key string      Path to TLS key file (default "/Users/ali/.docker/key.pem")
```

### Note for community:
Please 👍 if you'd like to upvote this or receive updates.

**Comment by njculver** [comment]
Trying to hack through this one - https://github.com/njculver/infracost/tree/feat/tls-configuration

**Comment by alikhajeh1** [comment]
@aliscott any thoughts on the [interface changes](https://github.com/njculver/infracost/commit/76ab6109c0bafd9778ccb8106f9a602e5be033ee) for this?

**Comment by aliscott** [comment]
Sorry I dropped the ball on this one. Have been looking at other tools - here's the options they have:

**Docker**

```
--tls                Use TLS; implied by --tlsverify
--tlscacert string   Trust certs signed only by this CA (default "/Users/ali/.docker/ca.pem")
--tlscert string     Path to TLS certificate file (default "/Users/ali/.docker/cert.pem")
--tlskey string      Path to TLS key file (default "/Users/ali/.docker/key.pem")
--tlsverify          Use TLS and verify the remote
```

**Redis**

```
--tls              Establish a secure TLS connection.
--sni <host>       Server name indication for TLS.
--cacert <file>    CA Certificate file to verify with.
--cacertdir <dir>  Directory where trusted CA certificates are stored.
                    If neither cacert nor cacertdir are specified, the default
                    system-wide trusted root certs configuration will apply.
--insecure         Allow insecure TLS connection by skipping cert validation.
--cert <file>      Client certificate to authenticate with.
--key <file>       Private key file to authenticate with.
--tls-ciphers <list> Sets the list of prefered ciphers (TLSv1.2 and below)
                    in order of preference from highest to lowest separated by colon (":").
                    See the ciphers(1ssl) manpage for more information about the syntax of this string.
--tls-ciphersuites <list> Sets the list of prefered ciphersuites (TLSv1.3)
                    in order of preference from highest to lowest separated by colon (":").
                    See the ciphers(1ssl) manpage for more information about the syntax of this string,
                    and specifically for TLSv1.3 ciphersuites.
--raw              Use raw formatting for replies (default when STDOUT is
```

**Helm**

```
--ca-file string             verify certificates of HTTPS-enabled servers using this CA bundle
--cert-file string           identify HTTPS client using this SSL certificate file
--insecure-skip-tls-verify   skip tls certificate checks for the repository
--key-file string            identify HTTPS client using this SSL key file
```

For us I think it makes sense to make these `infracost configure` values initially instead of flags since that where we set the `pricing_api_endpoint`. Here's my idea for the names we should use:

* `tls_insecure_skip_verify`
* `tls_ca_cert_file`
* `tls_cert_file`
* `tls_key_file`

@njculver, @alikhajeh1 do we even need to support the client cert and key files initially or is it enough initially to skip the client verification and just allow specifying a CA cert file?

**Comment by alikhajeh1** [comment]
Nice one! `infracost configure` makes sense, I suppose it can also be used in CI/CD pipelines too.
I don't have an opinion if we can skip the client verification part - if @njculver doesn't need it and it's a lot more work then sure 🤷

**Comment by aliscott** [comment]
@njculver I've opened a PR for adding `tls_insecure_skip_verify`, `tls_ca_cert_file`: https://github.com/infracost/infracost/pull/1187.

Let me know what you think and if this will help your use-case.

**Comment by njculver** [comment]
Sorry for letting this completely fall off my radar - I've been super busy with other things. Personally, I don't need the client certificate validation for my use case either. Skipping verification seems like it's fine for POC/dev usage, allowing a user to provide the CA cert works for self/internal signed certs. The additional client verification would probably be necessary in some more tightly controlled environments but this would be a good starting point from my perspective.

**Comment by aliscott** [comment]
Great, thanks @wilkej and @njculver, we'll ship this in the next release (v0.9.15) which we'll look to release at the end of this week/early next week. I'll let you know when it's released if you'd be willing to try it out!

**Comment by aliscott** [comment]
@wilkej, @njculver the support for `tls_insecure_skip_verify` and `tls_ca_cert_file` is now released in v0.9.15. Docs: https://www.infracost.io/docs/cloud_pricing_api/self_hosted/#using-a-self-signed-certificate. Please try it out and let me know if it works well for you 😄
